modules/core/src/lib/attribute-manager.js

```diff
@@ -282,8 +283,13 @@ export default class AttributeManager {
     for (const attributeName in attributes) {
       const attribute = attributes[attributeName];
       if (attribute.needsRedraw({clearChangedFlags: true})) {
-        // Only return non-transition attributes
-        if (!attributeTransitionManager.hasAttribute(attributeName)) {
+        if (attribute.userData.shaderAttributes) {
+          this._getChangedShaderAttributes(
```

Comment:
`Object.assign(changedAttributes, this._getChangedShaderAttributes());`

See comment below

Reply:
I'd actually prefer to inline the logic from `_getChangedShaderAttributes`, as it isn't going to be used elsewhere, but the linter complains that it becomes too nested. Any chance we'd consider relaxing that rule?

Reply:
`// eslint-disable-next-line max-depth` (or whatever rule you are breaking) is your friend in this case.

Reply:
Ah, that's handy, thanks!